### ISSUE TITLE
Differential expression

### ISSUE THREAD

**Issue by victorsanchezarevalo** [mousepixels/sanbomics_scripts#11]
Hi Mark,

I have a question. I am doing differential expression between two cell types in scRNA-seq, I tried scVI and it worked but diffxpy did not worked, it gives me this error: ZeroDivisionError: float division by zero

I wrote them with an issue, apparently it is a common error because others described the same problem. I am waiting for the reply but they did not to respond.

I wanted to performed this method, becase I am worried about the discrepancy of DEGs that I have with scVI (approximately 2400 DEGs) with the Differential expression using Scanpy function using Wilconson test (9.000 DEGs). Have you compared different methods for differential expression? Could you give me any advice, please?.

Thanks a lot.

Victor

**Comment by mousepixels** [comment]
Yeah I have had similar issues with diffxpy too. Make sure you are not passing the scaled X.. if not then it is an issue with diffxpy not you. 

I haven't done a side-by-side comparison of the two. However, I think that I would trust the results from scvi more by default. However, did you subset on variable genes before training the scvi model? If so, there could be genes left out.

**Comment by victorsanchezarevalo** [comment]
Thanks Mark

I run the not scaled X and I had the same error. I think it is an issue with diffxpy. 

With scVI I followed your tutorial and I did not subset the data, I used the adata. I had 2400 DEGs compared with 9000 with Scanpy Wilconson test.

I really appreciate your help!

Thanks!

Victor

**Comment by mrcrag1** [comment]
Hi, thanks for posting all the great material!
I'm running through your PyDeseq2 notebook and you pass a 'clinical' argument when you create the DeseqDataSet. This doesn't compile for me, and I can find nothing when I RTM (pydeseq2.readthedocs.io).
Thanks,
mrcrag

**Comment by gaozili1989** [comment]
I got this error as well: ZeroDivisionError: float division by zero

**Comment by mousepixels** [comment]
> Hi, thanks for posting all the great material! I'm running through your PyDeseq2 notebook and you pass a 'clinical' argument when you create the DeseqDataSet. This doesn't compile for me, and I can find nothing when I RTM (pydeseq2.readthedocs.io). Thanks, mrcrag

They changed clinical to metadata.

**Comment by mousepixels** [comment]
I don't really use diffxpy much anymore because it has these little unaddressed issues. I mostly do psuedobulk with pydesq2 now
